### DISCUSSION TITLE
FS totalbytes / usedbytes not included?

### DISCUSSION THREAD

**Discussion by holgerlembke** [espressif/arduino-esp32#8523] [General]
Moin.

Is there any reason why totalbytes() and usedbytes() is not included into the FS class?

I ended with writing 
```
uint64_t Gedoens::totalBytes(fs::FS *fs)
{
  if (fs == &SD_MMC)
  {
    return SD_MMC.totalBytes();
  }
  else if (fs == &SD)
  {
    return SD.totalBytes();
  }
  else if (fs == &LittleFS)
  {
    return LittleFS.totalBytes();
  }
  else if (fs == &FFat)
  {
    return FFat.totalBytes();
  }
  else
  {
    return -1;
  }
}
```

which is not too pretty and of course will fail if another file system is introduced.

Or say it differently: is there any reason that I should not implement it and then the pull request will be accepted?

mfg

**Comment by me-no-dev** [comment]
was not part of all APIs in the beginning. We could look into adding it there if all support it.
